src/System.Management.Automation/engine/CommandPathSearch.cs

```diff
@@ -110,7 +110,17 @@ private void ResolveCurrentDirectoryInLookupPaths()
                 sessionState.CurrentDrive.Provider.NameEquals(fileSystemProviderName) &&
                 sessionState.IsProviderLoaded(fileSystemProviderName);
 
-            string environmentCurrentDirectory = Directory.GetCurrentDirectory();
```

Comment:
It seems `Directory.GetCurrentDirectory()` is used at 4 places in the code base from my quick search:
https://github.com/PowerShell/PowerShell/search?q=Directory.GetCurrentDirectory%28%29&type=code

I guess when the process working directory is gone, the other 3 uses would throw when their code paths are hit. Can you please review and update those uses accordingly? Thanks

Reply:
- this PR handles the first case in CommandPathSearch.cs
- InitialSessionState.cs already handles if `GetCurrentDirectory()` fails on start and uses `$PSHome`
- Parser.cs is using it for a `using assembly` with a relative path, so we have to let it fail if CWD no longer exists
- InitialSessionStateProvider.cs is also on startup for assembly loading so should be fine

I think the 3 cases are fine